bin/prepare_for_multiqc.py

```diff
@@ -0,0 +1,170 @@
+    psm = unassigned_peptide_df.loc[:, "sequence"].value_counts()
+    df["psm"] = df["sequence"].map(psm)
+    return df
+
+
+def process_file(file, prefix, mass_range, prec_charge, fdr_threshold, quantify, keep_cols):
+    # If quantification is enabled, parse multiTSV output, otherwise the TSV file already has the correct format
+    if quantify == 'true':
+        data = parse_multiTSV(file)
+        n_psms = np.sum(data["psm"])
+    else:
+        data = pd.read_csv(file, sep='\t')
+        n_psms = 0
+
+    # Filter the columns down to a user-defined subset of columns
+    if keep_cols:
+        missing_columns = set(keep_cols) - set(data.columns)
+        if missing_columns:
+            raise ValueError(f"The following columns do not exist in the DataFrame: {missing_columns}")
+        data = data.loc[:, keep_cols]
+
+    # Check if all required columns are present in the DataFrame
+    required_columns = ['sequence', 'accessions', 'mz', 'observed_retention_time_best', 'score', 'COMET:xcorr']
+    missing_columns = set(required_columns) - set(data.columns)
+    if missing_columns:
+        raise ValueError(f"The following required columns are missing: {missing_columns}")
+
+    # Extract the length of the peptides.
+    # Remove everything inside parentheses, including the parentheses.
+    seq_length = data["sequence"].apply(lambda seq: len(re.sub(r'\(.*?\)', '', seq)))
+    seq_length = dict(Counter(seq_length))
+    with open(f"{prefix}_peptide_length.tsv", "w") as f:
+        for length, count in seq_length.items():
+            f.write(f"{length}\t{count}\n")
+
+    # General statistics
+    n_peptides = len(set(data["sequence"]))
+    n_modified_peptides = sum(1 for s in set(data["sequence"]) if '(' in s)
+    # Split the accession codes and count each protein accession individually
+    n_proteins = len(set([protein for entry in data["accessions"] for protein in entry.split(';')]))
+    with open(f"{prefix}_general_stats.tsv", "w") as f:
+        f.write(f"Sample\t# Peptides\t# Modified Peptides\t# Proteins\t# PSMs\n")
+        f.write(f"{prefix}\t{n_peptides}\t{n_modified_peptides}\t{n_proteins}\t{n_psms}\n")
+
+    histograms = [[data["mz"].astype(float), f"{prefix}_histogram_mz.tsv"],
+                  [data["observed_retention_time_best"].astype(float), f"{prefix}_histogram_rt.tsv"],
+                  [data["score"].astype(float), f"{prefix}_histogram_scores.tsv"],
+                  [data["COMET:xcorr"].astype(float), f"{prefix}_histogram_xcorr_scores.tsv"]]
+
+    for values, title in histograms:
+        hist, bin_edges = np.histogram(values, bins='auto')
+        with open(title, "w") as f:
+            for i in range(len(bin_edges) - 1):
+                bin_midpoint = (bin_edges[i] + bin_edges[i + 1]) / 2
+                f.write(f'{bin_midpoint}\t{hist[i]}\n')
+
+    data.to_csv(f"{prefix}_final.tsv", sep='\t', index=False)
+
+
+def main():
```

Comment:
What happens if this script gets an empty file tsv file (e.g. only the headers?). If its not too messy, I would suggest returning e.g. general statistics where each column value is 0. By this the user knows that there is nothing in the output. I think all other plots can ignore empty files. You can simulate an empty file (hopefully) by setting a very harsh `--fdr_threshold` of 0.0000001 or so.